pkg/file/splitter/internal/job.go

```diff
@@ -129,31 +139,45 @@ func (s *SimpleSplitterJob) sumLevel(lvl int) ([]byte, error) {
 
 	sizeToSum := s.cursors[lvl] - s.cursors[lvl+1]
 
-	// perform hashing
+	var ref encryption.Key
+	var chunkData []byte
+	data := s.buffer[s.cursors[lvl+1] : s.cursors[lvl+1]+sizeToSum]
```

Comment:
This can be done after SetSpan and avoid the allocation in case of an error. Also I am not sure why ref and chunkData declarations are moved here.